ui/modal/info_modal.go

```diff
@@ -171,22 +183,28 @@ func (in *InfoModal) HandleKeyEvent(evt *key.Event) {
 
 func (in *InfoModal) Handle() {
 	for in.btnPositve.Clicked() {
-		in.DismissModal(in)
 		isChecked := false
 		if in.checkbox.CheckBox != nil {
 			isChecked = in.checkbox.CheckBox.Value
 		}
 
 		in.positiveButtonClicked(isChecked)
+		if !in.isLoading {
+			in.DismissModal(in)
+		}
```

Comment:
ideally, i will expect this `isLoading` check to be at the top just before line 186. 
This way, nothing else get executed when there is a process running.
```
if in.isLoading{
     return
}
```

Reply:
This has been a point of confusion, so I'll try to explain in more detail.
https://github.com/planetdecred/godcr/blob/5f9db205405319305acce5ba8b7818a66d7f6ea9/ui/modal/info_modal.go#L192-L194 cannot be placed above https://github.com/planetdecred/godcr/blob/5f9db205405319305acce5ba8b7818a66d7f6ea9/ui/modal/info_modal.go#L191 because this is the called back that's passed to https://github.com/planetdecred/godcr/blob/5f9db205405319305acce5ba8b7818a66d7f6ea9/ui/modal/info_modal.go#L123

Think about it this way. When the positive button is clicked, an event is generated and https://github.com/planetdecred/godcr/blob/5f9db205405319305acce5ba8b7818a66d7f6ea9/ui/modal/info_modal.go#L184 is called. At this point https://github.com/planetdecred/godcr/blob/5f9db205405319305acce5ba8b7818a66d7f6ea9/ui/modal/info_modal.go#L47 is false because the call back passed to https://github.com/planetdecred/godcr/blob/5f9db205405319305acce5ba8b7818a66d7f6ea9/ui/modal/info_modal.go#L123 hasn't been executed. At some point https://github.com/planetdecred/godcr/blob/5f9db205405319305acce5ba8b7818a66d7f6ea9/ui/modal/info_modal.go#L191 -the callback- is executed and https://github.com/planetdecred/godcr/blob/5f9db205405319305acce5ba8b7818a66d7f6ea9/ui/modal/info_modal.go#L47 is true. The consequence of placing that logic above the callback is that the modal will always dismiss before execution is completed - try it!!!. 

The way I figured to prevent double execution is to check if https://github.com/planetdecred/godcr/blob/5f9db205405319305acce5ba8b7818a66d7f6ea9/ui/modal/info_modal.go#L47 is true twice first at the start of https://github.com/planetdecred/godcr/blob/5f9db205405319305acce5ba8b7818a66d7f6ea9/ui/modal/info_modal.go#L184 execution and second after https://github.com/planetdecred/godcr/blob/5f9db205405319305acce5ba8b7818a66d7f6ea9/ui/modal/info_modal.go#L191 -the callback- is executed. That's the point I was trying to get across to you earlier.

Other option is to rewrite https://github.com/planetdecred/godcr/blob/5f9db205405319305acce5ba8b7818a66d7f6ea9/ui/modal/info_modal.go#L123-L127

Reply:
Alright. I think i get your point... you are trying to prevent the modal from dismissing when a process is still running.

Originally, the `positiveButtonClicked` callback just waits for the executable codes in the `PositiveButton()` where ever it is used to complete executing and send a feedback to the button to dismiss.

What we have currently are 2 methods running side by side and really does not need one to finish before the second one starts.

Before now, it was just an info modal and really was not designed what what we used it for. Hence the reason for the initial implementation as the `positiveButtonClicked` tend to execute completely before the modl dismisses.

That not withstanding, what we needed to have done was have the `positiveButtonClicked` function like that on the password modal. Have it return a `bool` value on then does the modal dismiss.

But we should check the `isLoading` at the top level on button click and return.

```
if positiveButtonClicked() {
   Dismiss()
}
```

Reply:
That was my thought initially but I have to keep the task withing scope. I'll rewrite the callback to return a bool.